opm/material/fluidsystems/GenericOilGasFluidSystem.hpp

```diff
@@ -99,9 +99,8 @@ namespace Opm {
             {}
         };
 
-        static bool phaseIsActive(unsigned phaseIdx)
+        static bool phaseIsActive(int phaseIdx)
```

Comment:
We have a lot of places in our code base that assume the `phaseIdx` is an `unsigned int`.  This change will just push the conversion to the callers instead.

Reply:
It is the original `assert(phaseIdx < numPhases);` causing problem with `waterPhaseIdx == -1`. 

since we remove the assertion, we can continue to use the unsigned here.  And ` `waterPhaseIdx == -1` needs to be addressed at a later stage.